src/key_gen/tests.rs

```diff
@@ -256,29 +257,41 @@ fn threshold_signature() {
         })
         .collect();
 
-    // Test a threshold signature
-    let sig = pub_key_set
-        .combine_signatures(sig_shares.iter().take(THRESHOLD + 1))
-        .expect("signature shares match");
-    assert!(pub_key_set.public_key().verify(&sig, msg));
+    // Test threshold signature verification for a combination of signatures
+    let sig_combinations = sig_shares.iter().combinations(THRESHOLD + 1);
 
-    // Test a second threshold signature
-    let sig2_start_idx = rng.gen_range(1, std::cmp::max(2, sig_shares.len()));
-    let sig2 = pub_key_set
-        .combine_signatures(
-            sig_shares
-                .iter()
-                .cycle()
-                .skip(sig2_start_idx)
-                .take(THRESHOLD + 1),
-        )
-        .expect("signature shares match");
-    assert!(pub_key_set.public_key().verify(&sig2, msg));
+    let deficient_sig_combinations = sig_shares.iter().combinations(THRESHOLD - 1);
```

Comment:
Wouldn't it make more sense to test with `THRESHOLD` instead of `THRESHOLD - 1` because it's the maximum number of shares that is still not enough?